toolchains/utils.cmake

```diff
@@ -22,6 +22,19 @@ function(check_unsupported_compiler_version)
 
 endfunction(check_unsupported_compiler_version)
 
+function(check_mapzen_api_key)
+
+    if("${MAPZEN_API_KEY}" STREQUAL "")
```

Comment:
This syntax will check for a definition that was explicitly passed into CMake, which our `Makefile` currently does, but this doesn't automatically work if you're not using the `Makefile` to build (e.g. I use `ninja` almost exclusively on desktop). We can check for environment variables directly in CMake using `$ENV{MAPZEN_API_KEY}` and then this will work without the `Makefile`. (Comment moved for clarity)